server/src/main/java/org/elasticsearch/env/NodeMetadata.java

```diff
@@ -217,6 +237,42 @@ public NodeMetadata fromXContent(XContentParser parser) throws IOException {
         }
     }
 
+    /**
+     * Determine an index version checkpoint from a version ID
+     *
+     * <p>IndexVersion IDs diverged from Version IDs in 8.11.0. The 8.11 and 8.12 lines will
+     * write Version IDs in their node metadata, so we have to convert those IDs to specific
+     * IndexVersions. In 8.10 and earlier, IndexVersion and Version IDs are aligned. In 8.13 and
+     * later, NodeMetadata should write an IndexVersion to disk as a checkpoint.
+     *
+     * @param versionId a version ID as defined in {@link Version} and {@link IndexVersion}
+     * @return IndexVersion
+     */
+    private static IndexVersion idToIndexVersionCheckpoint(int versionId) {
+        // case -- ids match
+        Version version = Version.fromId(versionId);
```

Comment:
Would `VersionLookup` in https://github.com/elastic/elasticsearch/pull/105141 help here?